File: msme_intel-main/msme-demo/lib/types/ceed.ts
// lib/types/ceed.ts
export interface CEEDAnalysis {
  id: string;
  company_id: string;
  sector_id: string;
  
  // Quadrant Scores (0-100)
  core_score: number;
  expansion_score: number;
  efficiency_score: number;
  disruption_score: number;
  
  primary_quadrant: 'core' | 'expansion' | 'efficiency' | 'disruption';
  secondary_quadrant: 'core' | 'expansion' | 'efficiency' | 'disruption';
  
  core_assessment: CEEDQuadrantAssessment;
  expansion_assessment: CEEDQuadrantAssessment;
  efficiency_assessment: CEEDQuadrantAssessment;
  disruption_assessment: CEEDQuadrantAssessment;
  
  automation_readiness_score: number;
  automation_priority_level: 'critical' | 'high' | 'medium' | 'low';
  estimated_automation_roi: number;
  
  analyzed_at: string;
  confidence_score: number;
}

export interface CEEDQuadrantAssessment {
  score: number;
  sub_scores: Record<string, number>;
  evidence: string[];
  gaps: string[];
  opportunities: string[];
  automation_opportunities?: {
    area: string;
    current_state: string;
    automation_potential: string;
    estimated_impact: string;
  }[];
  disruptive_opportunities?: {
    opportunity: string;
    potential_impact: string;
    barriers: string[];
  }[];
}

File: msme_intel-main/backend/app/models/ceed.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime

class CEEDQuadrantAssessment(BaseModel):
    score: int
    sub_scores: Dict[str, int]
    evidence: List[str]
    gaps: List[str]
    opportunities: List[str]
    automation_opportunities: Optional[List[Dict[str, Any]]] = None
    disruptive_opportunities: Optional[List[Dict[str, Any]]] = None

class CEEDAnalysisBase(BaseModel):
    company_id: str
    sector_id: str
    core_score: int
    expansion_score: int
    efficiency_score: int
    disruption_score: int
    primary_quadrant: str
    secondary_quadrant: Optional[str] = None
    core_assessment: Dict[str, Any]
    expansion_assessment: Dict[str, Any]
    efficiency_assessment: Dict[str, Any]
    disruption_assessment: Dict[str, Any]
    automation_readiness_score: int
    automation_priority_level: str
    estimated_automation_roi: float
    confidence_score: float

class CEEDAnalysisCreate(CEEDAnalysisBase):
    pass

class CEEDAnalysis(CEEDAnalysisBase):
    id: str
    analyzed_at: datetime
    created_at: datetime

    class Config:
        from_attributes = True
